### DISCUSSION TITLE
on_bar  on_calculate  on_calculate_done 这三个回调函数有啥区别

### DISCUSSION THREAD

**Discussion by puddingsmall** [wondertrader/wondertrader#49] [Q&A] [answered]
on_bar  on_calculate  on_calculate_done 这三个回调函数有啥区别

**Comment by ZzzzHeJ** [comment, marked as answer]
可以参考链接
[事件函数说明](https://zzzzhej.github.io/WonderTrader-Learning-Notes/%E5%BC%80%E5%8F%91%E6%89%8B%E5%86%8C/WTPY/1.%E4%BA%A4%E6%98%93%E5%BC%95%E6%93%8E/1.CTA%E5%BC%95%E6%93%8E/%E4%BA%8B%E4%BB%B6%E5%87%BD%E6%95%B0.html#on_calculate)
